doc/user/jruby-java-interop.md

```diff
@@ -0,0 +1,106 @@
+# JRuby-Compatible Java Interop
+
+JRuby provides a rich set of interop functionality between Ruby and Java, and we
+attempt to emulate it faithfully in TruffleRuby.
```

Comment:
I think it would be useful to mention early that this only works on JVM, not on SubstrateVM currently.